m2cgen/interpreters/javascript/interpreter.py

```diff
@@ -0,0 +1,61 @@
+import os
+
+from m2cgen import ast
+from m2cgen.interpreters import mixins
+from m2cgen.interpreters import utils
+from m2cgen.interpreters.interpreter import ToCodeInterpreter
+from m2cgen.interpreters.javascript.code_generator \
+    import JavascriptCodeGenerator
+
+
+class JavascriptInterpreter(ToCodeInterpreter,
+                            mixins.LinearAlgebraMixin,
+                            mixins.SubroutinesAsFunctionsMixin):
```

Comment:
Are you sure that JS code will benefit from splitting the generated code into subroutines? The initial purpose of splitting code into smaller functions was to workaround limitations related to the size of an individual function/method. Eg. JVM has strict constrains on how large the individual method can be - this is why we use subroutines for Java code. 

If there are no similar constrains in JS and if having smaller functions doesn't improve the runtime, I'd suggest to remove subroutine support to make the implementation simpler.

Reply:
I'm glad to remove the subroutine support.  I honestly wasn't sure what `SubroutinesAsFunctions` was doing and just copied the Java implementation to start.

I'll add the e2e tests, and remove the `SubroutinesAsFunctionsMixin`.  Let me know what you want done with the util functions.